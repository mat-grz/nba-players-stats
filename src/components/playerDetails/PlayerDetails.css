.player-info-container {
    display: flex;
    justify-content: space-between;
}

.player-info {
    text-align: left;
}

.player-name {
    color: #BBE1FA;
    margin-bottom: 0;
}

.player-team {
    color: #3282B8;
    margin-top: 0;
}