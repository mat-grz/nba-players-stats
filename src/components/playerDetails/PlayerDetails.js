import React from 'react';
import PropTypes from 'prop-types';
import './PlayerDetails.css';

function PlayerDetails({ first_name, last_name, team = 'No team' }) {
    return (
        <div className="player-info-container">
            <div className="player-info">
                <h1 className="player-name">{first_name} {last_name}</h1>
                <h3 className="player-team">{team}</h3>
            </div>
            <img src="shirt.svg" alt="shirt" />
        </div>
    );
}

PlayerDetails.propTypes = {
    first_name: PropTypes.string.isRequired,
    last_name: PropTypes.string.isRequired,
    team: PropTypes.string
};

export default PlayerDetails;