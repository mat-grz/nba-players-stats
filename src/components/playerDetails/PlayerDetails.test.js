import React from 'react';
import { shallow } from 'enzyme';
import PlayerDetails from './PlayerDetails';

const props = {
    first_name: 'James',
    last_name: 'Harden',
    team: 'Houston Rockets'
};

const spy = jest.spyOn(global.console, 'error');

it('should display correct player info', () => {

    const component = shallow(<PlayerDetails {...props} />);
    expect(component).toMatchSnapshot();
});

it('should display player info without a team', () => {
    const { team, ...playerWithoutTeam } = props;
    const component = shallow(<PlayerDetails {...playerWithoutTeam}/>);

    expect(component.contains('No team')).toEqual(true);
});

it('first_name and last_name should be required', () => {
    shallow(<PlayerDetails />);

    expect(spy).toBeCalledTimes(2);
});