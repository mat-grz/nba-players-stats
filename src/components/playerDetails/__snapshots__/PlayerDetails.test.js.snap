// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correct player info 1`] = `
<div
  className="player-info-container"
>
  <div
    className="player-info"
  >
    <h1
      className="player-name"
    >
      James
       
      Harden
    </h1>
    <h3
      className="player-team"
    >
      Houston Rockets
    </h3>
  </div>
  <img
    alt="shirt"
    src="shirt.svg"
  />
</div>
`;
